common/determined_common/api/request.py

```diff
@@ -12,18 +16,40 @@
 from determined_common.api import authentication, errors
 
 # The path to a file containing an SSL certificate to trust specifically for the master, if any, or
-# False to disable cert verification entirely.
-_master_cert_bundle = None
+# False to disable cert verification entirely. If set to a path, it should always be a temporary
+# file that we own and can delete.
+_master_cert_bundle = None  # type: Optional[Union[str, bool]]
 
 # The name we use to verify the master.
 _master_cert_name = None
 
 
+@atexit.register
```

Comment:
Just a thought, might not be worth it, but can we `@atexit.register` a closure bound to the `path` string inside `_set_master_bundle()`?  Maybe give us an iota more of protection in multithreading situations.

Reply:
Okay, I think I like that. It means that there will be multiple files sitting around and only removed on exit if `set_master_cert_bundle` is called multiple times, but the logic is cleaner and multiple calls should be rare.